src/services/refactors/convertArrowFunction.ts

```diff
@@ -0,0 +1,90 @@
+/* @internal */
+namespace ts.refactor.convertArrowFunction {
+    const refactorName = "Convert arrow function";
+    const refactorDescription = Diagnostics.Convert_arrow_function.message;
+    const addBracesActionName = "Add braces to arrow function";
+    const removeBracesActionName = "Remove braces from arrow function";
+    const addBracesActionDescription = Diagnostics.Add_braces_to_arrow_function.message;
+    const removeBracesActionDescription = Diagnostics.Remove_braces_from_arrow_function.message;
+    registerRefactor(refactorName, { getEditsForAction, getAvailableActions });
+
+    interface Info {
+        container: ArrowFunction;
+        expression: Expression;
+        addBraces: boolean;
+    }
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+        const info = getConvertibleArrowFunctionAtPosition(file, startPosition);
+        if (!info) return undefined;
+
+        const actions: RefactorActionInfo[] = [
+            info.addBraces ?
+                {
+                    name: addBracesActionName,
+                    description: addBracesActionDescription
+                } : {
+                    name: removeBracesActionName,
+                    description: removeBracesActionDescription
+                }
+        ];
+
+        return [{
+            name: refactorName,
+            description: refactorDescription,
+            actions
+        }];
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+        const info = getConvertibleArrowFunctionAtPosition(file, startPosition);
+        if (!info) return undefined;
+
+        const { addBraces, expression, container } = info;

```

Comment:
Is it possible `addBraces` could disagree with `_actionName`.  If that does happen, should we bail?